### PR TITLE
Add Typecheck Prerequisites

### PR DESCRIPTION
**Pull request by kbader-godaddy**
## Summary

- Updated `eslint-config-godaddy` to latest
- Added types and typechecking for `@gasket/fetch`
- Added typecheck prerequisites for `@gasket/helper-intl`
- Added typecheck prerequisites for `@gasket/nextjs`
- Added typecheck prerequisites for `@gasket/plugin-docs-graphs`
- Added typecheck prerequisites for `@gasket/plugin-docs`
- Added typecheck prerequisites for `@gasket/plugin-docusaurus`
- Added typecheck prerequisites for `@gasket/plugin-elastic-apm`
- Added typecheck prerequisites for `@gasket/plugin-fastify`
- Added typecheck prerequisites for `@gasket/plugin-git`
- Added typecheck prerequisites for `@gasket/plugin-intl`
- Added typecheck prerequisites for `@gasket/plugin-lint`
- Added typecheck prerequisites for `@gasket/plugin-manifest`
- Added typecheck prerequisites for `@gasket/plugin-nextjs`
- Added typecheck prerequisites for `@gasket/plugin-redux`
- Added typecheck prerequisites for `@gasket/plugin-service-worker`
- Added typecheck prerequisites for `@gasket/plugin-webpack`
- Added typecheck prerequisites for `@gasket/plugin-workbox`
- Added typecheck prerequisites for `@gasket/react-intl`
- Added typecheck prerequisites for `@gasket/redux`
- Added typecheck prerequisites for `@gasket/utils`
- Small tweaks to align-package script

## Test Plan

Existing tests pass with new `typecheck` scripts.

### CODE REVIEW

**Comment by kbader-godaddy** [comment]
> ## `package-lock.json` changes
> Click to toggle table visibility
> 
> Name	Status	Previous	Current
> `typescript`	[DOWNGRADED](#)	5.4.3	5.1.6

This was done because of a warning message we are receiving from the typescript tests package.

File: package.json [hunk 1]
```diff
@@ -57,7 +57,8 @@
     "nx": "^15.0.13",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
-    "rimraf": "^3.0.2"
+    "rimraf": "^3.0.2",
+    "typescript": "5.1.6"
```
Comment (agerard-godaddy) [inline package.json:61]: Why is typescript pinned to 5.1.6 in this PR? You have `^5.4.2` in the open source repo ([example](https://github.com/godaddy/gasket/blob/main/packages/gasket-plugin-analyze/package.json#L55))

Comment (kbader-godaddy) [reply package.json:61]: I'm happy to continue to use latest version of typescript, but didn't want to go breaking the typescript-eslint.

Comment (agerard-godaddy) [reply package.json:61]: If it is just a warning and still works, given we can hopefully just remove `gasket-typescript-tests` once these typechecks are added directly to each package, I'd prefer to see us keep up with latest if possible.

Comment (kbader-godaddy) [reply package.json:61]: 10-4

**Review by agerard-godaddy — APPROVED** [review]

**Pull request merged**